enhancements/cluster-logging/flow-control-api.md

```diff
@@ -0,0 +1,164 @@
+
+This proposal defines an API to let cluster administrators to limit logging rates, or ignore some logs entirely.
+Logs will still be lost if the collector cannot keep up, but administrators have more control over *what* is lost.
+
+Rate limits mean that:
+
+* The cost and volume of logging can be predicted more accurately in advance.
+* Noisy containers cannot produce unbounded log traffic and unfairly drown out other containers.
+* High-value logs can be preferred over low-value logs by assigning higher rate limits.
+* Log streams that are not needed can be ignored, which reduces the load on the logging infrastructure.
+
+The API allows selected rates and policies to be applied by namespaces or pod labels.
+New types of selector that are added in future should also be applicable.
+
+## Motivation
+
+### Goals
+
+### Non-Goals
+
+The following are *not* goals for this proposal but *may* be addressed in future proposals:
+
+- `block` policy: containers that exceed their rate limit are back-pressured by leaving their stdout/stderr unread until they are within the target rate.
+- combined rate limits: specify a *combined* rate limit, i.e. limit the sum of all container rates logging to the same pipeline. Dynamically adjust individual rate limits to give an approximately equal individual limit to each container.
+
+## Proposal
+
+Add a new CR `ClusterLogFlowControl`:
+
+`spec`: []FlowControlRule
+
+FlowControlRule: Match all criteria (AND)
+- `logType`: (enum: application|infrastructure|audit, default: application)
+- `namespaces`: ([]string, optional) policy applies to containers in any of these namespaces.
+- `selector`: ([LabelSelector][LabelSelector])
+- `maxByteRate`: ([Quantity][Quantity], default 0) bytes per second in [k8s resource notation][memory]
+  Absent or <= 0 means no limit (other than the limits of the logging system).
+  Use `policy: ignore` to ignore logs, not `maxByteRate: 0`.
+- `policy`: (enum: ignore|drop|block, default: drop)
+  - `ignore`: no logs are forwarded, if possible they are not even collected.
+  - `drop`: drop data if necessary to respect `maxByteRate`, or if the logging system cannot keep up.
+
+If multiple rules match a container the *last* matching rule applies.
+You can specify general rules followed by more specific over-ride rules.
+For example a default rule for a namespace, followed by an over-ride for pods with specific labels in the namespace.
+
+[LabelSelector]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#labelselector-v1-meta
+[quantity]: https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/
+[memory]: (https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory)
+
+### User Stories
+
+#### Cluster admin wants to set global per-container default of 1024 bytes/second
+
+``` yaml
+- maxByteRate: 1024
+```
+
+#### Cluster admin wants to ignore logs from containers with certain labels
```

Comment:
Is that the scenario where you could basically "exclude" collecting logs from certain systems or are you still collecting but you don't do anything?

Reply:
Exclude collecting, unless there's some compelling reason why we can't (don't know of any right now)